stock_picking_auto_revert/models/stock_picking.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2019 ForgeFlow
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models, exceptions, _
+from odoo.tools import float_compare
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    @api.multi
+    def action_revert_recreate(self):
+        self.ensure_one()
+        pick = self
+        msg = _(
+            "This picking cannot be returned because the products "
+            "are not available in the destination location."
+        )
+        # Create return picking
+        StockReturnPicking = self.env["stock.return.picking"]
+        default_data = StockReturnPicking.with_context(
+            active_ids=pick.ids, active_id=pick.id
+        ).default_get(
+            [
+                "move_dest_exists",
+                "original_location_id",
+                "product_return_moves",
+                "parent_location_id",
+                "location_id",
+            ]
+        )
+        for rm in default_data["product_return_moves"]:
```

Comment:
In order to get better visibility and to allow overriding to maybe complete conditions for a picking to be returned, I would have done this computation in another function. And better a computed field 'can_be_returned' or 'products_available_for_return'. What do you think ?